zkevm-circuits/src/exp_circuit.rs

```diff
@@ -278,84 +288,93 @@ impl<F: Field> SubCircuitConfig<F> for ExpCircuitConfig<F> {
-                                region.assign_advice(
-                                    || format!("exp circuit: {:?}: {}", *column, offset + i),
-                                    *column,
-                                    offset + i,
-                                    || Value::known(*value),
-                                )?;
-                            }
-                        }
-                        offset += OFFSET_INCREMENT;
-                    }
+    fn assign_exp_event(
+        &self,
+        region: &mut Region<F>,
+        offset: &mut usize,
+        exp_event: &ExpEvent,
+        mul_chip: &mut MulAddChip<F>,
+        parity_check_chip: &mut MulAddChip<F>,
+    ) -> Result<(), Error> {
+        let mut exponent = exp_event.exponent;
+        for (step, step_assignments) in exp_event
+            .steps
+            .iter()
+            .rev()
+            .zip(ExpTable::assignments::<F>(exp_event).chunks_exact(OFFSET_INCREMENT))
+        {
+            // assign everything except the exp table.
+            self.assign_step(
+                region,
+                *offset,
+                &mut exponent,
+                step,
+                mul_chip,
+                parity_check_chip,
+            )?;
+            // assign exp table.
+            for (i, assignment) in step_assignments.iter().enumerate() {
+                for (column, value) in self.exp_table.columns().iter().zip(assignment) {
+                    region.assign_advice(
+                        || format!("exp circuit: {:?}: {}", *column, *offset + i),
+                        *column,
+                        *offset + i,
+                        || Value::known(*value),
+                    )?;
                 }
-
-                self.assign_padding_rows(&mut region, offset)?;
-
-                Ok(())
-            },
-        )
+            }
+            region.assign_fixed(
+                || format!("exp_circuit: {:?}: {}", self.exp_table.is_step, offset),
+                self.exp_table.is_step,
+                *offset,
+                || Value::known(F::one()),
+            )?;
+            for i in 1..OFFSET_INCREMENT {
+                region.assign_fixed(
+                    || format!("exp_circuit: {:?}: {}", self.exp_table.is_step, *offset + i),
+                    self.exp_table.is_step,
+                    *offset + i,
+                    || Value::known(F::zero()),
+                )?;
+            }
+            // mul_chip has 7 rows, exp_table has 4 rows. So we increment the offset by
+            // the maximum number of rows taken up by any gadget within the
+            // exponentiation circuit.
+            *offset += OFFSET_INCREMENT;
+        }
+        Ok(())
     }
 
-    fn assign_padding_rows(&self, region: &mut Region<'_, F>, offset: usize) -> Result<(), Error> {
+    fn assign_unused_rows(&self, region: &mut Region<'_, F>, offset: usize) -> Result<(), Error> {
```

Comment:
I don't think it will matter much, but I would personally prefer explicitly also assigning `0` to the fixed column `is_step`.

If it's later made a complex selector, then no need. But because its a fixed column at the moment, it should be assigned to avoid any "unassigned" errors that may pop up later on.